shared/modules/metametrics.js

```diff
@@ -111,6 +115,12 @@ export const segment = process.env.SEGMENT_WRITE_KEY
  * @property {string}  [currency] - ISO 4127 format currency for events with revenue, defaults to US dollars
  * @property {number}  [value] - Abstract "value" that this event has for MetaMask.
  * @property {boolean} [excludeMetaMetricsId] - whether to exclude the user's metametrics id for anonymity
+ * @property {string}  [metaMetricsId] - an override for the metaMetricsId in the event one is created as part
```

Comment:
I'm not sure I follow this part - it looks like `metaMetricsId` should always be set, even during the "opt-in" event :thinking: 

Did you find any cases where it wasn't set?

Reply:
Oh, I think I see now, during the opt-in flow. Well this is quite silly.

The entire flow of `setParticipateInMetaMetrics` does use Promises properly and it blocks until everything has been set, so you'd think the ID should be available by then. But it doesn't do a background sync before returning. There's even a special method used to set the flag in the `metamask` redux slice immediately, and the `metaMetricsId` is in-scope, but it's not set :facepalm:.

Well, this is easily fixed. I'll give this a shot in a separate PR, so that maybe we can remove all this special handling here.

Reply:
Welp. So I got something working along these lines, but it didn't work as well as I thought it would. Probably because of the React render lifecycle - even if Redux is in sync with the background, in the component we'll still have the old copy until a new render finishes.

Reply:
When I get to this part in the new event schema would you be down to walk through how we could improve the flow? I'd love to remove this option as it introduces some convolution in the logic.

Reply:
I think how you have this setup is close enough to being the best we'll get so long as metrics are tracked in React components. A simpler solution might be to dispatch events directly in Redux actions instead - that would get around this React lifecycle issue.